external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaWriter.scala

```diff
@@ -94,4 +94,10 @@ private[kafka010] object KafkaWriter extends Logging {
       }
     }
   }
+
+  def close(sc: SparkContext, kafkaParams: ju.Map[String, Object]): Unit = {
+    sc.parallelize(1 to 10000).foreachPartition { iter =>
+      CachedKafkaProducer.close(kafkaParams)
+    }
```

Comment:
Using guave cache, we can close if not used for a certain time. Shall we ignore closing them during a shutdown ? 
In the particular case of kafka producer, I do not see a direct problem with that. Since we do a producer.flush() on each batch. I was just wondering, with streaming sinks general - what should be our strategy ?